### ISSUE TITLE
Bug: `EXPERIMENTAL_useProjectService: true` throws unclear error if file is not included in a program

### ISSUE THREAD

**Issue by mrazauskas** [typescript-eslint/typescript-eslint#7435]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Repro

To the root of a project that is using `EXPERIMENTAL_useProjectService: true` recently I added `rollup.config.js`. That file was not added to tsconfig's `include` list. ESLint throws: 

> Error while loading rule '@typescript-eslint/consistent-type-exports': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.

By the way, it was so exciting to see the `EXPERIMENTAL_useProjectService` option added. The project I talk about is rather experimental so it was perfect playground to tryout the `ProjectService`.

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  parserOptions: {
    "EXPERIMENTAL_useProjectService": true
  }
};
```

### tsconfig

```jsonc
{
  "include": ["./src"]
}
```

### Expected Result

Perhaps the error could be more useful. If this should be an error at all. Not sure what is going on.

### Actual Result

 Looking at the error, I thought that maybe the option was removed or stoped working.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/parser`            | `6.2.1` |

**Comment by JoshuaKGoldberg** [comment]
Ahh so exciting that you're trying out the new option @mrazauskas! 🙌 

Would you be able to provide us with a minimum reproduction of what you're seeing locally? Failing that, would we be able to see the original code? We're happy to sign whatever NDAs are needed if this is private.

**Comment by mrazauskas** [comment]
Reproduction is here: https://github.com/mrazauskas/x-use-project-service

1. Install with `yarn`
2. Run `yarn lint` and you will see the error
3. Delete `rollup.config.js` file in the root
4. Run `yarn lint` and the error is gone
5. Bring `rollup.config.js` file back
6. Open `tsconfig.json` and uncomment one line in the `include` list
7. Run `yarn lint`, there will be no error as well

**Comment by probablyup** [comment]
I ran into this too today.

```
Error: Error while loading rule '@typescript-eslint/dot-notation': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
Occurred while linting /Users/esj/code/frontend-v2/packages/cad-pro/src/integration/autodesigner/autodesigner-defaults.docker-integration.tsx
    at getParserServices (/Users/esj/code/frontend-v2/node_modules/@typescript-eslint/utils/dist/eslint-utils/getParserServices.js:24:15)
    at create (/Users/esj/code/frontend-v2/node_modules/@typescript-eslint/eslint-plugin/dist/rules/dot-notation.js:84:55)
    at Object.create (/Users/esj/code/frontend-v2/node_modules/@typescript-eslint/utils/dist/eslint-utils/RuleCreator.js:38:20)
    at createRuleListeners (/Users/esj/code/frontend-v2/node_modules/eslint/lib/linter/linter.js:870:21)
    at /Users/esj/code/frontend-v2/node_modules/eslint/lib/linter/linter.js:1040:110
    at Array.forEach (<anonymous>)
    at runRules (/Users/esj/code/frontend-v2/node_modules/eslint/lib/linter/linter.js:977:34)
    at Linter._verifyWithoutProcessors (/Users/esj/code/frontend-v2/node_modules/eslint/lib/linter/linter.js:1330:31)
    at Linter._verifyWithConfigArray (/Users/esj/code/frontend-v2/node_modules/eslint/lib/linter/linter.js:1706:21)
    at Linter.verify (/Users/esj/code/frontend-v2/node_modules/eslint/lib/linter/linter.js:1412:65)
v%    
```

My config (override object):

```
files: ['*.{ts,tsx}'],
settings: {
  'import/parsers': {
    '@typescript-eslint/parser': ['.ts', '.tsx', '.d.ts'],
  },
  'import/resolver': {
    node: {
      extensions: ['.js', '.jsx', '.json', '.d.ts', '.tsx', '.ts', '.mjs'],
    },
  },
},
parser: '@typescript-eslint/parser',
parserOptions: {
  project: ['./tsconfig.json', './tsconfig.integration.json'],
  EXPERIMENTAL_useProjectService: true,
},
extends: [
  'plugin:@typescript-eslint/eslint-recommended',
  'plugin:@typescript-eslint/recommended',
],
plugins: [
  '@typescript-eslint',
],
```

The file being linted is covered by `./tsconfig.integration.json` but it seems like that `project` settings field might be ignored now?

**Comment by mrazauskas** [comment]
@JoshuaKGoldberg It feels like this issue is resolved after #8815. Or?

**Comment by JoshuaKGoldberg** [comment]
Yes, good call - thanks! #8815 also resolved this.
